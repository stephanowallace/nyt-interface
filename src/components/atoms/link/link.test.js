import React from 'react';
import { shallow } from 'enzyme';
import Link from '.';

describe('Link', () => {
  it('renders correctly', () => {
    let component = shallow(<Link href="test">test</Link>);
    expect(component).toMatchSnapshot();

    component = shallow(<Link href="test" visited>test</Link>);
    expect(component).toMatchSnapshot();
  });

  it('renders child in an anchor tag', () => {
    const component = shallow(<Link href="test">test</Link>);
    expect(component.find('a').text()).toBe('test');
  });

  it('shows specific class when set as visited', () => {
    const component = shallow(<Link href="test" visited>test</Link>);
    expect(component.find('.visited')).toHaveLength(1);
  });
});
