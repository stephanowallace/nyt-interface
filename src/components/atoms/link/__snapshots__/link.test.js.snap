// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders correctly 1`] = `
<a
  className="
      link
      
      "
  href="test"
  target="_blank"
>
  test
</a>
`;

exports[`Link renders correctly 2`] = `
<a
  className="
      link
      visited
      "
  href="test"
  target="_blank"
>
  test
</a>
`;
